### DISCUSSION TITLE
virtual4c assigning nan to point of view

### DISCUSSION THREAD

**Discussion by claudiarabu** [open2c/cooltools#443] [General]
Hi!

I was using the virtual4c function of cooltools and noticed that the point of view is assigned as nan, but this might lead to some confusion as also null (= not happening) interactions are assigned as nan. Also, for plotting I substituted every nan with 0 values, but the resulting graph at around the pov bins was of course a bit misleading. 
What I did was changing the cooltools/api/virtual4c.py to assign value 1 to the pov bins, but I guess another option could be to assign 0 to null interactions and nan to the pov bins.

Besides that, cool tools indeed! :)

**Comment by Phlya** [comment]
Mh can you clarify in some more detail why setting bins within the viewpoint to Nan is misleading? I think that should at least be the default behavior...

**Reply by claudiarabu** [reply]
I find misleading the fact that both viewpoint bins and non-interacting bins have the same value (nan) when their behavior is kind of opposite - that is: interaction at the highest level (as it corresponds to the heat map diagonal) and no interaction at all, respectively.
Maybe I am wrong though!

**Reply by Phlya** [reply]
Ok, actually when there simply aren't any interactions it should be 0, not Nan... NaN would be for the case of masked bins.

And I feel like masking bins inside the viewpoint is necessary, otherwise I'm not sure what should go there... Imagine you have a viewpoint 5 bins long?

**Reply by claudiarabu** [reply]
Yes, I agree that assigning no interactions to 0 and nan to the pov bins would be the best option!

P.s. I still have nan in case of no interactions, but maybe that was not intended and it caused the discussion!

**Reply by Phlya** [reply]
Are you sure it's for the case of no interactions and not for the case of masked bins?
If so, it's a bug...

**Reply by claudiarabu** [reply]
Well, yes, unfortunately I am sure and I guess it's a bug.

**Reply by Phlya** [reply]
Ok, thank you, I'll look into it next week then! On holidays until then.

**Reply by claudiarabu** [reply]
Hi Phyla,
Long story short: it's not a bug. I needed to manipulate my sample before converting it to mcool to adjust the interaction values of some regions with a different ploidy and this caused many bins to be considered as masked in the virtual4c, but if I don't manipulate the sample I do get 0s and nan as separate values. Sorry for the confusion and enjoy your holidays!
